### PR TITLE
Feature/jc19 479 anime list management react

### PR DESCRIPTION
**Pull request by SofyaKirillova**

### CODE REVIEW

File: apps/react/src/api/services/anime.ts [hunk 1]
```diff
@@ -1,32 +1,51 @@
...
-    const nextUrl = useAppSelector(selectAnimeNextUrl).replace(CONFIG.apiUrl, '');
-    if (nextUrl === null) {
-      const { data } = await http.get<PaginationDto<AnimeCommonDto>>(nextUrl);
-      return PaginationMapper.fromDto(data, AnimeCommonMapper.fromDto);
+  /**
+   * Fetches a list of anime.
+   * @param params Anime params.
+   */
+  export async function fetchAnimeList(params?: AnimeQueryParams): Promise<AnimeCommon[]> {
+    const url = new URLSearchParams();
```
Comment (artem-sotnichenko-saritasa) [inline apps/react/src/api/services/anime.ts:25]: It's not a url, consider better name.

File: apps/react/src/features/anime/components/AnimeListManagement/AnimeListManagement.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  'airedStart',
+] as const;
+
+const typeList: readonly AnimeType[] = Object.values(AnimeType);
+
+/** Anime list management page component. */
+const AnimeListManagementComponent: FC = () => {
+  const [params, setParams] = useSearchParams();
+  const dispatch = useAppDispatch();
+  const getDefaultAnimeParams = useRef<AnimeParams>({
```
Comment (artem-sotnichenko-saritasa) [inline apps/react/src/features/anime/components/AnimeListManagement/AnimeListManagement.tsx:34]: I think "initial" is more correct than "default".
Also I suggest to move out default values like 'titleEnglish' into named variables.

File: apps/react/src/features/anime/components/AnimeListManagement/AnimeListManagement.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  const handleChangeType = (event: SelectChangeEvent<typeof types>) => {
+    const {
+      target: { value },
+    } = event;
+    setTypes(typeof value === 'string' ? value.split(',') : value);
+  };
+
+  const [ordering, setOrdering] = useState(
+    getDefaultAnimeParams.current.ordering,
+  );
```
Comment (artem-sotnichenko-saritasa) [inline apps/react/src/features/anime/components/AnimeListManagement/AnimeListManagement.tsx:52]: You could set proper types with generics, then you don't need to cast type on 69th line

File: apps/react/src/features/anime/pages/AnimePage/AnimePage.tsx [hunk 4]
```diff
@@ -1,41 +1,45 @@
...
 
-  useEffect(() => {
-    dispatch(fetchAnimeList());
-  }, [dispatch]);
-
   return (
     <Grid container spacing={3}>
-      <Grid item xs={4}>
-        <List className={styles['anime-list']}>
+      <Grid className={styles['dashboard-Anime-films']} item xs={4} sx={{ maxHeight: '100vh' }}>
```
Comment (artem-sotnichenko-saritasa) [inline apps/react/src/features/anime/pages/AnimePage/AnimePage.tsx:30]: Please don't mix styling with css(className) and props(sx), choose one for consistency

**Review by artem-sotnichenko-saritasa — APPROVED** [review]

**Review by golovachev-yaroslav — APPROVED** [review]

**Pull request closed without merging**